doc/threshold/SONiC Threshold feature spec.md

```diff
@@ -0,0 +1,589 @@
+	  * [2.2 Functional Description](#2_2-functional-description)
+  * [3 Design](#3-design)
+	  * [3.1 Overview](#3_1-overview)
+		  * [3.1.1 ThresholdMgr](#3_1_1-thresholdmgr)	
+	  * [3.2 DB Changes](#3_2_db-changes)  
+		  * [3.2.1 CONFIG DB](#3_2_1-config-db)
+		  * [3.2.2 APP DB](#3_2_2-app-db)
+		  * [3.2.3 STATE DB](#3_2_3-state-db)
+		  * [3.2.4 ASIC DB](#3_2_4-asic-db)
+		  * [3.2.5 COUNTER DB](#3_2_5-counter-db)
+	  * [3.3 Switch State Service Design](#3_3-switch-state-service-design)
+		  * [3.3.1 Orchestration Agent](#3_3_1-orchestration-agent)
+		  * [3.3.2 Other Process](#3_3_2-other-process)
+	  * [3.4 Syncd](#3_4-syncd)
+	  * [3.5 SAI](#3_5-sai)
+	  * [3.6 CLI](#3_6-cli)
+		  * [3.6.1 Data Models](#3_6_1-data-models)
+		  * [3.6.2 Configuration Commands](#3_6_2-configuration-commands)
+		  * [3.6.3 Show Commands](#3_6_3-show-commands)
+		  * [3.6.4 Clear Commands](#3_6_4-clear-commands)
+		  * [3.6.5 Debug Commands](#3_6_5-debug-commands)
+		  * [3.6.6 REST API Support](#3_6_6-rest-api-support)  
+  * [4 Flow Diagrams](#4-flow-diagrams)
+	  * [4.1 Config Call Flow](#4_1-config-call-flow)
+	  * [4.2 Breach Event Call Flow](#4_2-breach-event-call-flow)
+  * [5 Error Handling](#5-error-handling)
+  * [6 Serviceability And Debug](#6-serviceability-and-debug)
+  * [7 Warm Boot Support](#7-warm-boot-support)
+  * [8 Scalability](#8-scalability)
+  * [9 Unit Test](#9-unit-test)
+  * [10 Internal Design Information](#10-internal-design-information)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 06/12/2019  |   Shirisha Dasari  | Initial version                   |
+
+# About this Manual
+This document provides general information about the threshold feature implementation in SONiC.
+
+# Scope
+This document describes the high level design of threshold feature. 
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| SAI                      | Switch abstraction interface        |
+| TAM                      | Telemetry and monitoring            |
+| Watermark                | Peak counter value reached          |
+| ThresholdMgr             | Threshold manager                   |
+
+# 1 Feature Overview
+The threshold feature allows configuration of a threshold on supported buffers in ingress and egress. A threshold breach notification (entry update in COUNTERS\_DB) is generated on breach of the configured buffer threshold in ASIC.   
+
+## 1.1 Requirements
+### 1.1.1 Functional Requirements
+
+1.0 Threshold feature allows the user to configure a threshold on a buffer (i.e buffer usage) and record all threshold breach events for that buffer.
```

Comment:
Does the spec expose the ASIC capabilities to the user whether the ASIC capable of threshold feature?

Reply:
Hi,
There is a PR submitted by Vissu with regard to exposing ASIC's capabilities, https://github.com/opencomputeproject/SAI/pull/989. This issue can get addressed as a part of the same PR.